plasma_framework/contracts/mocks/framework/registries/VaultRegistryMock.sol

```diff
@@ -0,0 +1,15 @@
+pragma solidity ^0.5.0;
+
+import "../../../src/framework/registries/VaultRegistry.sol";
+
+contract VaultRegistryMock is VaultRegistry {
+    bool public vaultCheckPass;
+
+    constructor() public {
+        vaultCheckPass = false;
+    }
+
+    function checkOnlyFromVault() public onlyFromVault {
+        vaultCheckPass = true;
```

Comment:
What about emitting an event here? The checks could be made repeatedly.

Reply:
Sounds like a good idea. Would fix this Monday.

Reply:
fixed